lib/linter/linter.js

```diff
@@ -45,6 +45,7 @@ const DEFAULT_ECMA_VERSION = 5;
 const commentParser = new ConfigCommentParser();
 const DEFAULT_ERROR_LOC = { start: { line: 1, column: 0 }, end: { line: 1, column: 1 } };
 const parserSymbol = Symbol.for("eslint.RuleTester.parser");
+const merge = Object.assign;
```

Comment:
We can't replace here `lodash.merge` with `Object.assign` , because `merge()` is used to merge `parserOptions`, which should be deeply merged. I added a test case that would fail with `Object.assign` in https://github.com/eslint/eslint/pull/14948.

Reply:
Guess we can close this PR then cuz Object.assign don't cut it...